src/lib.rs

```diff
@@ -752,10 +720,15 @@ impl<K: EnrKey> Enr<K> {
                 *self = enr_backup;
                 return Err(EnrError::UnsupportedIdentityScheme);
             }
-            inserted.push(
-                self.content
-                    .insert(key.as_ref().to_vec(), rlp::encode(&(value)).freeze()),
-            );
+
+            let value = rlp::encode(&(value)).freeze();
```

Comment:
Should `remove_insert` `insert_key_values`'s Item.1 also be generic over `T: Encodable`?

Reply:
Yeah, but it's a bit out of scope for the current PR. I'm also thinking we can improve this api in a different way later on